### ISSUE TITLE
Support more granular env var config for API server

### ISSUE THREAD

**Issue by sbellem** [bigchaindb/bigchaindb#575]
### Currently

`BIGCHAINDB_SERVER_BIND` and `BIGCHAINDB_API_ENDPOINT` can be used to set the http server socket binding and the api endpoint url, as documented in [server.bind, server.workers & server.threads](https://bigchaindb.readthedocs.io/en/latest/server-reference/configuration.html?highlight=api_endpoint#server-bind-server-workers-server-threads) and [api_endpoint](https://bigchaindb.readthedocs.io/en/latest/server-reference/configuration.html?highlight=api_endpoint#api-endpoint).
### Limitations

In the case of `server.bind` one has to pass both both the host and port, and in the case of the api endpoint one has to pass the entire url, which is not convenient for client code, in which one may only need to control the host, for instance.
### Proposal

This ticket wishes to add more granular control, such that one can set the host, and port (at least) separately, and fall back on default values for the rest.

In other words, the necessary environment variables are provided so that the server binding location and api endpoint url can be constructed. E,g,:

``` bash
BIGCHAINDB_SERVER_BIND_HOST
BIGCHAINDB_SERVER_BIND_PORT
BIGCHAINDB_API_HOST
BIGCHAINDB_API_PORT
BIGCHAINDB_API_SCHEME
BIGCHAINDB_API_VERSION
```
